### DISCUSSION TITLE
Create selector list with stacked modifiers

### DISCUSSION THREAD

**Discussion by AlanBreck** [tailwindlabs/tailwindcss#12953] [Help] [answered]
~Is it possible to create variants that when stacked they create a selector list instead of a compound selector?~

[UPDATE on account of [this comment](https://github.com/tailwindlabs/tailwindcss/discussions/12953#discussioncomment-8473484)]: Is it possible to create variants that when stacked build on each other as sibling modifiers rather than making one modifier the descendant of the other?

For example, given a config like this:

```js
const plugin = require('tailwindcss/plugin')

/** @type {import('tailwindcss').Config} */
export default {
  theme: {
    extend: {
      // ...
    },
  },
  plugins: [
    plugin(function ({ addVariant }) {
      addVariant('chrome', '.is-chrome &')
      addVariant('mobile', '.is-mobile &')
    }),
  ],
}
```

...and the following HTML:

```html
<div class="is-chrome is-mobile">
  <div class="hidden chrome:mobile:flex">
    ...
  </div>
</div>
```

[UPDATED] ...the resulting selector would be something like

```
.is-mobile.is-chrome .chrome\:mobile\:flex
```

...instead of

```
.is-chrome .is-mobile .chrome\:mobile\:flex
```

**Comment by wongjn** [comment]
You'd need to create a another variant that combines the two, like:
```js
addVariant('chrome-mobile', ['.is-chrome &', '.is-mobile &']);
// or
addVariant('chrome-mobile', ':is(.is-chrome, .is-mobile) &');
```
```html
<div class="is-chrome is-mobile">
  <div class="hidden chrome-mobile:flex">
```
See this working in [this Tailwind Play](https://play.tailwindcss.com/ezAeq3dlec).

Or use an [arbitrary variant](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) like:

```html
<div class="is-chrome is-mobile">
  <div class="hidden [:is(.is-chrome,.is-mobile)_&]:flex">
```
See this working in [this Tailwind Play](https://play.tailwindcss.com/rnYui1KMLG).

**Reply by AlanBreck** [reply]
Thanks for the reply!

Yeah, I figured those two could be options. There's no way to avoid explicitly defining those combinations, though?

**Reply by wongjn** [reply]
Not that I know of.

**Reply by adamwathan** [reply, marked as answer]
There is! Do this instead:

```
const plugin = require('tailwindcss/plugin')

/** @type {import('tailwindcss').Config} */
export default {
  theme: {
    extend: {
      // ...
    },
  },
  plugins: [
    plugin(function ({ addVariant }) {
      addVariant('chrome', '&:is(.is-chrome *)')
      addVariant('mobile', '&:is(.is-mobile *)')
    }),
  ],
}
```

**Reply by adamwathan** [reply]
It’s not quite what’s in the original post though, but I’m wondering if maybe it is the behaviour you’re actually looking for. Using a selector list approach, that element in your example would be visible on mobile OR in Chrome, and my guess is you are actually trying to test if something is mobile AND in Chrome.

**Reply by AlanBreck** [reply]
🤦 Yep, that's exactly what I was actually wanting to do.

This is perfect! Thank you!
